src/actions/index.js

```diff
@@ -334,9 +335,8 @@ export const showDeleteProjectPrompt = (project: Project) => ({
   project,
 });
 
-export const showModal = (modal: string) => ({
-  type: SHOW_MODAL,
-  modal: modal,
+export const showProjectSettings = () => ({
+  type: SHOW_PROJECT_SETTINGS,
```

Comment:
I was debating saying something about this during the original review; I prefer when actions are descriptive of the thing being done, not its effect. A user isn't "showing a modal", they're "showing the project settings".

I didn't say anything in the review because it wasn't a big deal, but I realized something: we get better Flow type safety when we do it this way.

Before, state.modal could be any string, because the showModal action could set it to any value. Now, the type is only 1 of 2 possible strings, because there are 2 explicit actions that change it.

I also removed a related comment below, since I don't think we need a MODAL constant - instead we can rely on Flow for that

Reply:
Ah, OK, thanks for the explanation to the action naming. I'll use this convention in the future.
One question, is `HIDE_MODAL` action name OK or should we rename it to `HIDE_PROJECT_SETTINGS`? 

You're right. It's OK to not have the modal constant as flow will complain about wrong modal strings.

Reply:
Yeah, I thought about that myself. I think it's OK, since if the user is clicking the "X", they actually do want to dismiss the modal; that's the action they're taking.

Maybe `DISMISS_MODAL` would be a bit more semantic, but I don't think we need to create a new action for every modal, since the action the user is taking is always the same - "get this modal out of my way"